smart_contract/derive/Cargo.toml

```diff
@@ -14,6 +14,9 @@ workspace = true
 proc-macro = true
 
 [dependencies]
-syn.workspace = true
-quote.workspace = true
-proc-macro2.workspace = true
+iroha_macro_utils = { workspace = true }
+
+syn2 = { workspace = true }
```

Comment:
Should we name it `syn` instead of `syn2`, if we already don't support `syn` version 1 in this crate? Or we going to rename it after migrating in all crates?

Reply:
Is there a way to rename a workspace dependency? If there is, I think that it would make sense to use it here

Reply:
Using [`package`](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml) won't work?

Reply:
Nope, renaming deps inherited from the workspace doesn't seem to be implemented as of now https://github.com/rust-lang/cargo/issues/12546